lib/metrics/default-metrics.js

```diff
@@ -0,0 +1,175 @@
+'use strict';
+
+const ClientMetrics = require('./client-metrics');
+const EventEmitter = require('events');
+
+/**
+ * A default implementation of [ClientMetrics]{@link module:metrics~ClientMetrics} that exposes the driver events as
+ * Node.js events.
+ * <p>
+ *   An instance of [DefaultMetrics]{@link module:metrics~DefaultMetrics} is configured by default in the client,
+ *   you can access this instance using [Client#metrics]{@link Client#metrics} property.
+ * </p>
+ * @implements {module:metrics~ClientMetrics}
+ * @alias module:metrics~DefaultMetrics
+ * @example <caption>Listening to events emitted</caption>
+ * defaultMetrics.errors.on('increment', err => totalErrors++);
+ * defaultMetrics.errors.clientTimeout.on('increment', () => clientTimeoutErrors++);
+ * defaultMetrics.speculativeRetries.on('increment', () => specExecsCount++);
+ * defaultMetrics.successfulResponses.on('increment', latency => specExecsCount++);
+ */
+class DefaultMetrics extends ClientMetrics {
```

Comment:
This is a neat design because it almost serves as a Dual API.   If a user wants, they can just subscribe on the event emitters, or they can implement their own `ClientMetrics` interface.